java/src/main/java/com/google/cloud/dataproc/templates/util/ReadSchemaUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.dataproc.templates.util;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.StructType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReadSchemaUtil {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ReadSchemaUtil.class);
+
+  public static StructType readSchema(String schemaUrl)  {
```

Comment:
add unit tests for same

Reply:
Skipping Test for Util class that reads GCS bucket. Require explicit GCS object creation to test functionality